### DISCUSSION TITLE
Block device support

### DISCUSSION THREAD

**Discussion by ibonny** [seaweedfs/seaweedfs#3808] [Ideas]
Is there any way to get block device support through seaweed? I love the filesystem and object store support so far. I'm using it for a distributed fs (very easy to set up) for Proxmox, but I do have a desire to use block devices back by it.

**Comment by chrislusf** [comment]
This is not started yet. Need to understand how a block storage system works.

**Comment by ibonny** [comment]
Let me know if there is anything I can do to help. I can probably figure out the block/object store mapping, but not the device driver to support the low-level block device.

**Reply by chrislusf** [reply]
Basically SeaweedFS can provide chunk => file_id storage. I am thinking to format the allocated disk space into small (e.g. 2MB) section, and the metadata would have an array of file_ids. I am not sure when a block storage system starts, how does it find some initial "entry point" to those metadata.

**Reply by ibonny** [reply]
Ok no problem.
